foundry/foundry.py

```diff
@@ -485,9 +486,9 @@ def download(self, globus=True, verbose=False, **kwargs):
             task_list = list(recursive_ls(self.transfer_client,
```

Comment:
If you want to be fancy, don't use a list first. That way you can download while finding the files (I'm assuming recursive ls is a generator)